### PR TITLE
doc: describe how to create a patch release

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-common/221)

### CODE REVIEW

File: doc/cutting-a-patch.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Creating a new patch release of google-cloud-cpp-common
+
+The intended audience of this document are developers in the
+`google-cloud-cpp-common` project that need to create a new release. The
```
Comment (devbww) [inline doc/cutting-a-patch.md]: s/release/patch release/?

File: doc/cutting-a-patch.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+```bash
+git clone git@github.com:${GITHUB_USERNAME}/google-cloud-cpp-common.git
+cd google-cloud-cpp-common
+git remote add upstream https://github.com/googleapis/google-cloud-cpp-common.git
+git fetch upstream
+```
+
+Now you can create a branch of the release branch, we will use `v0.22.x` as an
+example, replace that with the right version:
```
Comment (devbww) [inline doc/cutting-a-patch.md]: s/example/example of the release branch/ perhaps?

File: doc/cutting-a-patch.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+We will assume that you have created a clone of your fork, as in:
+
+```bash
+git clone git@github.com:${GITHUB_USERNAME}/google-cloud-cpp-common.git
+cd google-cloud-cpp-common
+git remote add upstream https://github.com/googleapis/google-cloud-cpp-common.git
+git fetch upstream
+```
+
+Now you can create a branch of the release branch, we will use `v0.22.x` as an
```
Comment (devbww) [inline doc/cutting-a-patch.md]: s/branch, we/branch.  We/

File: doc/cutting-a-patch.md (hunk 2 above)
Comment (devbww) [inline doc/cutting-a-patch.md]: s/example, replace/example.  Replace/

File: doc/cutting-a-patch.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+git checkout -b prepare-v0.22.1-patch
+```
+
+Then cherry-pick the changes you want, without committing them
+
+```bash
+git cherry-pick -n 140c7e3d0ad06accd639e95164f5de2c4c82642e
+```
+
+Update the release notes to include these changes, commit these changes. You
```
Comment (devbww) [inline doc/cutting-a-patch.md]: s/changes, commit these changes/changes, and commit them/

File: doc/cutting-a-patch.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+git commit
+```
+
+Run the usual tests, push the changes and create a PR. You must use `v0.22.x`
+as the target branch of your PR.
+
+## Creating the release
+
+As of this writing the release script always cuts releases from `master`,
+create the release manually using the GitHub website is probably best.
```
Comment (devbww) [inline doc/cutting-a-patch.md]: s/create/so creating/

**Comment by coryan** [comment]
Thanks for the review, much better.  I made some more edits than you asked, PTAL.

File: doc/cutting-a-patch.md (hunk 5 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: Fixed.

File: doc/cutting-a-patch.md (hunk 4 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: Fixed.

File: doc/cutting-a-patch.md (hunk 2 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: See above.

File: doc/cutting-a-patch.md (hunk 3 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: See above.

File: doc/cutting-a-patch.md (hunk 2 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: Used some variables instead, please take a look.

File: doc/cutting-a-patch.md (hunk 1 above)
Comment (coryan) [reply doc/cutting-a-patch.md]: Done.

**Pull request merged**